contrib/epee/src/portable_storage.cpp

```diff
@@ -39,6 +39,30 @@
 #include <string>
 #include <sstream>
 
+namespace
+{
+  /**
+   * @brief Takes a std::string by reference and wraps it with ostream-like access
+   */
+  struct string_ref_stream
+  {
+    string_ref_stream(std::string& str_ref): m_str_ref(str_ref) {}
+
+    void write(const char* buf, const std::size_t len)
+    {
+      m_str_ref.insert(m_str_ref.size(), buf, len);
```

Comment:
Trivial but this is : `m_str_ref.append(buf, len);`.

Reply:
>  with the downside that the entire epee binary serialization is re-generated for this additional type.


Yes I was worried about this too, but my monerod binary size only increased from 15512512 bytes to 15525032 bytes (+12.5 KB).